kubespawner/spawner.py

```diff
@@ -1537,6 +1538,16 @@ def _validate_image_pull_secrets(self, proposal):
           and the value is a dictionary with the following keys:
 
           - `display_name`: Name used to identify this particular option
+          - `other_choice`: Object to specify if there should be a free-form field if the user
+            select "Other" as a choice:
+            - `enabled`: Boolean, whether the free form input should be enabled
+            - `display_name`: String, label for input field
+            - `validation_regex`: Optional, regex that the free form input should match - eg. ^pangeo/.*$
+            - `validation_message`: Optional, validation message for the regex. Should describe the required
+               input format in a human-readable way.
+            - `kubespawner_override`: Object specifying what key:values should be over-ridden
```

Comment:
Specify that you can use `{value}` here to substitute whatever the user entered value was.